submitit/core/utils.py

```diff
@@ -294,25 +296,43 @@ def __call__(self, *args: Any, **kwargs: Any) -> str:
             env=self.env,
         ) as process:
             assert process.stdout is not None
+            assert process.stderr is not None
+            outno = process.stdout.fileno()
+            errno = process.stderr.fileno()
+            fds = [outno, errno]
+            targets: Dict[int, Tuple[List[str], IO[bytes], IO[str]]] = {
+                outno: (out_buffers, process.stdout, sys.stdout),
+                errno: (err_buffers, process.stderr, sys.stderr),
+            }
             try:
-                for line in iter(process.stdout.readline, b""):
-                    if not line:
-                        break
-                    outlines.append(line.decode().strip())
-                    if self.verbose:
-                        print(outlines[-1], flush=True)
+                while fds:
+                    # We use select to read either from stderr or stdout.
+                    # Failure to do so can result in a deadlock if the stderr
+                    # or the stdout buffer get overflown.
+                    # select is not the fastest, but it is the most supported.
+                    ready, _, _ = select.select(fds, [], [])
+                    for fd in ready:
+                        buffers, in_stream, out_stream = targets[fd]
+                        raw_buf = in_stream.read(2 ** 16)
```

Comment:
I no longer read entire lines, because a subprocess could potentially try to print one very big line, and the same old bug would come back.